src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -2317,7 +2318,45 @@ private void HandlePackageManagerWindowClosed(object sender, EventArgs e)
 
         internal void EnableEnvironment(bool isEnabled)
         {
-            this.mainGrid.IsEnabled = isEnabled;
+            this.mainGrid.IsEnabled = isEnabled;       
+        }
+
+        /// <summary>
+        /// Adds/Removes an overlay so the user won't be able to interact with the background (this behavior was implemented for Dynamo and for Library)
+        /// </summary>
+        /// <param name="isEnabled">True if the overlay is enable otherwise will be false</param>
+        internal void EnableOverlayBlocker(bool isEnabled)
+        {
+            object[] parametersInvokeScript = new object[] { "fullOverlayVisible", new object[] { isEnabled } };
+            ResourceUtilities.ExecuteJSFunction(this, parametersInvokeScript);
+
+            if (isEnabled)
+            {
+                //By default the shortcutsBarGrid has a ZIndex = 1 then will be shown over the overlay that's why we need to change the ZIndex
+                Panel.SetZIndex(shortcutsBarGrid, 0);
+                var backgroundElement = new GuideBackground(this)
+                {
+                    Name = "BackgroundBlocker",
+                    HorizontalAlignment = HorizontalAlignment.Left,
+                    VerticalAlignment = VerticalAlignment.Top,
+                    Visibility = Visibility.Visible
+                };
+
+                //adds the overlay to the main Dynamo grid
+                mainGrid.Children.Add(backgroundElement);
+                Grid.SetColumnSpan(backgroundElement, 5);
+                Grid.SetRowSpan(backgroundElement, 6);
```

Comment:
Are these span numbers fixed and could change after Dynamo view has more columns?

Reply:
@QilongTang fixed, the hardcoded values were removed in the next commit:https://github.com/DynamoDS/Dynamo/pull/14778/commits/c1e106652c8f543f7a0bf4fd132849158d55e907